src/MarkdownTextInput.web.tsx

```diff
@@ -1,22 +1,438 @@
+} catch (e) {
+  throw new Error('[react-native-live-markdown] Function `createReactDOMStyle` from react-native-web not found. Please make sure that you are using React Native Web 0.18 or newer.');
+}
+
+let preprocessStyle: (style: any) => any;
+try {
+  preprocessStyle =
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('react-native-web/dist/exports/StyleSheet/preprocess').default;
+} catch (e) {
+  throw new Error('[react-native-live-markdown] Function `preprocessStyle` from react-native-web not found.');
+}
+
+type MarkdownStyle = MarkdownTextInputDecoratorViewNativeComponent.MarkdownStyle;
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
 interface MarkdownTextInputProps extends TextInputProps {
-  // nothing here
+  markdownStyle?: MarkdownStyle;
+  onClick?: (e: MouseEvent<HTMLDivElement>) => void;
+  dir?: string;
+}
+
+interface MarkdownNativeEvent extends Event {
+  inputType: string;
+}
+
+type Selection = {
+  start: number;
+  end: number;
+};
+
+let focusTimeout: NodeJS.Timeout | null = null;
+
+// Removes one '\n' from the end of the string that were added by contentEditable div
+const normalizeValue = (value: string) => {
+  return value.replace(/\n$/, '');
+};
+
+// If an Input Method Editor is processing key input, the 'keyCode' is 229.
+// https://www.w3.org/TR/uievents/#determine-keydown-keyup-keyCode
+function isEventComposing(nativeEvent: globalThis.KeyboardEvent) {
+  return nativeEvent.isComposing || nativeEvent.keyCode === 229;
 }
 
-// eslint-disable-next-line arrow-body-style
-const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>((props, ref) => {
-  // TODO: add web implementation here
-  return (
-    <TextInput
-      {...props}
-      ref={ref}
-    />
-  );
+const getCurrentCursorPosition = CursorUtils.getCurrentCursorPosition;
+
+const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
+  (
+    {
+      accessibilityLabel,
+      accessibilityLabelledBy,
+      accessibilityRole,
+      autoCapitalize = 'sentences',
+      autoCorrect = true,
+      blurOnSubmit = false,
+      clearTextOnFocus,
+      dir = 'auto',
+      multiline = false,
+      markdownStyle,
+      onBlur,
+      onChange,
+      onChangeText,
+      onClick,
+      onFocus,
+      onKeyPress,
+      onSelectionChange,
+      onSubmitEditing,
+      placeholder = '\u200B', // Zero-width space to prevent div from collapsing
+      placeholderTextColor = '#ccc',
+      selectTextOnFocus,
+      spellCheck,
+      style = {},
+      value,
+    },
+    ref,
+  ) => {
+    const divRef = useRef<HTMLDivElement | null>(null);
+    const currentlyFocusedField = useRef<HTMLDivElement | null>(null);
+    const contentSelection = useRef<Selection | null>(null);
+    const className = `react-native-live-markdown-input-${multiline ? 'multiline' : 'single-line'}`;
+    const history = useRef(new MarkdownInputHistory(20));
```

Comment:
This syntax will create a new `MarkdownInputHistory` object on each call, how about this?
```suggestion
    const history = useRef(null);
    if (history.current === null) {
      history.current = new MarkdownInputHistory(20);
    }
```